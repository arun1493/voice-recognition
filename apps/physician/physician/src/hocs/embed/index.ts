export * from "./withEmbedAuth";
export * from "./withAmwellAuth";
