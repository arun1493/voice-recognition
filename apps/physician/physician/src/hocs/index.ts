export * from "./withIntercom";
export * from "./withAnalytics";
export * from "./withTestSetup";
export * from "./withBrowserSupport";
