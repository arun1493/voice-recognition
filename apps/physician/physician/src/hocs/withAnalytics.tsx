import { useLocation } from "react-router-dom";
import { FunctionComponent, useEffect } from "react";
import ReactGA, { FieldsObject, InitializeOptions } from "react-ga";

import { isDevelopmentEnvironment } from "../helpers/environment";

const ANALYTICS_TRACKING_ID = "UA-100205469-2";

const initializeAnalytics = (options: InitializeOptions = {}) => {
  ReactGA.initialize(ANALYTICS_TRACKING_ID, {
    debug: isDevelopmentEnvironment(),
    ...options
  });
};

function withAnalytics(WrappedComponent: FunctionComponent, options: FieldsObject = {}): FunctionComponent {
  return () => {
    const location = useLocation();

    const trackPage = () => {
      ReactGA.set({
        page: location.pathname,
        // TODO: Get userId from Redux
        userId: undefined,
        ...options
      });

      ReactGA.pageview(location.pathname);
    };

    useEffect(() => {
      trackPage();
    }, [location.pathname]);

    return <WrappedComponent />;
  };
}

export { initializeAnalytics, withAnalytics };
