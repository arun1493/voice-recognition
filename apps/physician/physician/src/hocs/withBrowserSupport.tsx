import _some from "lodash/some";
import { UAParser } from "ua-parser-js";
import { Paths } from "@suki-web/suki-business";
import { FunctionComponent, useEffect, useState } from "react";

import { getRoute } from "../helpers";
import { DEVICE_TYPES, Pages, SUPPORTED_BROWSERS } from "../constants";

function isSupportedBrowser(browserName: string, version: number | string): boolean {
  if (!(browserName in SUPPORTED_BROWSERS)) return false;
  if (typeof version === "string") {
    version = Number.parseInt(version);
  }

  return version >= SUPPORTED_BROWSERS[browserName];
}

function redirectUnsupportedBrowser(): void {
  const redirectRoute = getRoute(Pages.UnSupportedDeviceOrBrowser);
  if (redirectRoute) {
    window.location.href = redirectRoute;
  }
}

function getDeviceType() {
  const ua = navigator.userAgent;
  if (/(tablet|ipad|playbook|silk)|(android(?!.*mobi))/i.test(ua)) {
    return DEVICE_TYPES.tablet;
  }
  if (/Mobile|iP(hone|od)|Android|BlackBerry|IEMobile|Kindle|Silk-Accelerated|(hpw|web)OS|Opera M(obi|ini)/.test(ua)) {
    return DEVICE_TYPES.mobile;
  }
  return DEVICE_TYPES.desktop;
}

function isDesktop() {
  return getDeviceType() === DEVICE_TYPES.desktop;
}

function withBrowserSupport(WrappedComponent: FunctionComponent) {
  return () => {
    const [isBrowserSupported, setIsBrowserSupported] = useState<boolean>(false);

    const UNSUPPORTED_ROUTES: string[] = [
      Paths.AltaisEntry,
      Paths.Authentication,
      Paths.Embed,
      Paths.ForgotPassword,
      Paths.Help,
      Paths.Home,
      Paths.ImplicitCallback,
      Paths.IntroPage,
      Paths.Login,
      Paths.Patients,
      Paths.ScheduleImport,
      Paths.Settings,
      Paths.Settings,
      Paths.UnfinishedNotes,
      Paths.WindowsUpdate
    ];

    const isCurrentRouteUnSupported = (route: string) => {
      return window.location.pathname.includes(route);
    };

    const isUnsupportedRoute = _some(UNSUPPORTED_ROUTES, isCurrentRouteUnSupported);

    useEffect(() => {
      const parser = new UAParser();
      const browserInformation = parser.getBrowser();

      if (
        !isSupportedBrowser(browserInformation.name as string, browserInformation.version as string) ||
        (!isDesktop() && (isUnsupportedRoute || window.location.pathname === Paths.Root))
      ) {
        return redirectUnsupportedBrowser();
      }

      setIsBrowserSupported(true);
    }, []);

    return isBrowserSupported ? <WrappedComponent /> : null;
  };
}

export { withBrowserSupport };
