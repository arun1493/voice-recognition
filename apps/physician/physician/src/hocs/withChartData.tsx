import store from "store2";
import _isEmpty from "lodash/isEmpty";
import { FunctionComponent } from "react";
import { useDispatch } from "react-redux";
import { useQuery } from "@tanstack/react-query";
import { useLocation, useParams } from "react-router-dom";
import { EMPTY_OBJECT, Paths } from "@suki-web/suki-business";

import {
  getPatientAllergies,
  getPatientMedicalHistory,
  getPatientMedications,
  getPatientSurgicalHistory,
  getPatientVitals,
  logger
} from "../api";
import { noteActions } from "../redux/note";
import { getCardQueryParams } from "../helpers";
import { compositionReader, metadataReader } from "../readers";
import { CardType, LOCAL_STORE, QUERY_NAMES } from "../constants";

function withChartData(WrappedComponent: FunctionComponent<any>): FunctionComponent {
  return (props: any) => {
    const { composition, loading } = props;

    const dispatch = useDispatch();
    const location = useLocation();
    const isPatientProfilePage = location.pathname.includes(Paths.PatientProfile);
    const isEmrUser = store.get(LOCAL_STORE.IS_EMR_USER);

    let shouldFetchCardsData;
    let queryParams = EMPTY_OBJECT as any;

    if (isPatientProfilePage) {
      const { patientId = "" } = useParams<{ patientId: string }>();
      shouldFetchCardsData = !!(isEmrUser && patientId);
      queryParams = {
        sukiPatientId: patientId
      };
    } else {
      const metadata = compositionReader.metadata(composition);
      const patientInfo = metadataReader.patient(metadata);
      shouldFetchCardsData = (isEmrUser && !loading && !_isEmpty(composition) && !_isEmpty(patientInfo)) || false;
      queryParams = getCardQueryParams(composition);
    }

    // Allergies
    useQuery([QUERY_NAMES.PATIENT_ALLERGIES, queryParams], () => getPatientAllergies(queryParams), {
      enabled: shouldFetchCardsData,
      staleTime: 0,
      onError: (error) => logger.error({ message: "Error fetching allergies", payload: error }),
      onSuccess: (allergies) =>
        dispatch(noteActions.setCardsByType({ type: CardType.ALLERGIES, cards: allergies?.data || [] }))
    });

    // Medical History
    useQuery([QUERY_NAMES.PATIENT_MEDICAL_HISTORY, queryParams], () => getPatientMedicalHistory(queryParams), {
      enabled: shouldFetchCardsData,
      staleTime: 0,
      onError: (error) => logger.error({ message: "Error fetching medical history", payload: error }),
      onSuccess: (medicalHistory) =>
        dispatch(noteActions.setCardsByType({ type: CardType.PAST_MEDICAL_HISTORY, cards: medicalHistory?.data }))
    });

    // Medications
    useQuery([QUERY_NAMES.PATIENT_MEDICATIONS, queryParams], () => getPatientMedications(queryParams), {
      enabled: shouldFetchCardsData,
      staleTime: 0,
      onError: (error) => logger.error({ message: "Error fetching medications", payload: error }),
      onSuccess: (medications) =>
        dispatch(noteActions.setCardsByType({ type: CardType.CURRENT_MEDICATION, cards: medications?.data }))
    });

    // Surgical History
    useQuery([QUERY_NAMES.PATIENT_SURGICAL_HISTORY, queryParams], () => getPatientSurgicalHistory(queryParams), {
      enabled: shouldFetchCardsData,
      staleTime: 0,
      onError: (error) => logger.error({ message: "Error fetching surgical history", payload: error }),
      onSuccess: (surgicalHistory) =>
        dispatch(noteActions.setCardsByType({ type: CardType.PAST_SURGICAL_HISTORY, cards: surgicalHistory?.data }))
    });

    // Vitals
    useQuery([QUERY_NAMES.PATIENT_VITALS, queryParams], () => getPatientVitals(queryParams), {
      enabled: shouldFetchCardsData,
      staleTime: 0,
      onError: (error) => logger.error({ message: "Error fetching vitals", payload: error }),
      onSuccess: (vitals) => dispatch(noteActions.setCardsByType({ type: CardType.VITALS, cards: vitals?.data }))
    });

    return <WrappedComponent {...props} />;
  };
}

export { withChartData };
