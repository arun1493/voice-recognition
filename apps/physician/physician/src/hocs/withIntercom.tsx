import { FunctionComponent } from "react";
import { IntercomProvider } from "react-use-intercom";

import { isProductionEnvironment } from "../helpers/environment";

const INTERCOM_ID = isProductionEnvironment() ? "mrausu9h" : "tfrr9ikc";

function withIntercom(WrappedComponent: FunctionComponent): FunctionComponent {
  return () => {
    return (
      <IntercomProvider appId={INTERCOM_ID} autoBoot={true} autoBootProps={{ hideDefaultLauncher: true }}>
        <WrappedComponent />
      </IntercomProvider>
    );
  };
}

export { withIntercom };
