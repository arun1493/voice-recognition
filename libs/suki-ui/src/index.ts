export * from "./hooks";
export * from "./assets";
export * from "./helpers";
export * from "./lib/fab";
export * from "./lib/card";
export * from "./lib/icon";
export * from "./lib/link";
export * from "./lib/page";
export * from "./lib/text";
export * from "./lib/hooks";
export * from "./lib/modal";
export * from "./lib/theme";
export * from "./lib/button";
export * from "./lib/dialog";
export * from "./lib/header";
export * from "./lib/select";
export * from "./lib/tooltip";
export * from "./lib/checkbox";
export * from "./lib/sukiList";
export * from "./lib/constants";
export * from "./lib/formInput";
export * from "./lib/textField";
export * from "./lib/datePicker";
export * from "./lib/iconButton";
export * from "./lib/sukiButton";
export * from "./lib/pageHeader";
export * from "./lib/typography";
export * from "./lib/sukiTooltip";
export * from "./lib/dropDownMenu";
export * from "./lib/navigationIcon";
export * from "./lib/capturingIndicator";
