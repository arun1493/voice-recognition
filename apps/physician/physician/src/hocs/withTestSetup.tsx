import { Provider } from "react-redux";
import { lightTheme } from "@suki-web/suki-ui";
import { render } from "@suki-web/suki-business";
import { ThemeProvider } from "styled-components";
import { FunctionComponent, ReactNode } from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";

import { setupStore } from "../redux/store";

const queryClient = new QueryClient();

const componentSetup =
  (Component: FunctionComponent<any>, defaultProps = {}) =>
  (overriddenProps = {}) =>
    render(
      <Router>
        <ThemeProvider theme={lightTheme}>
          <QueryClientProvider client={queryClient}>
            <Component {...defaultProps} {...overriddenProps} />
          </QueryClientProvider>
        </ThemeProvider>
      </Router>
    );

const componentSetupWithStore =
  (getJSX: (defaultProps: any) => ReactNode) =>
  (overriddenProps = {}, preloadedState = {}) =>
    render(
      <Provider store={setupStore(preloadedState)}>
        <QueryClientProvider client={queryClient}>
          <Router>
            <ThemeProvider theme={lightTheme}>{getJSX(overriddenProps)}</ThemeProvider>
          </Router>
        </QueryClientProvider>
      </Provider>
    );

const componentWithTheme = (Component: FunctionComponent<any>, props = {}) => (
  <ThemeProvider theme={lightTheme}>
    <Component {...props} />
  </ThemeProvider>
);

const componentWithThemeAndStore = (Component: FunctionComponent<any>, props = {}, preloadedState = {}) => (
  <Provider store={setupStore(preloadedState)}>
    <QueryClientProvider client={queryClient}>
      <Router>
        <ThemeProvider theme={lightTheme}>
          <Component {...props} />
        </ThemeProvider>
      </Router>
    </QueryClientProvider>
  </Provider>
);

export { componentSetup, componentSetupWithStore, componentWithTheme, componentWithThemeAndStore };
