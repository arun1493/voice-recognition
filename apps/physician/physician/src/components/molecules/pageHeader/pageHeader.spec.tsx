import "@testing-library/jest-dom/extend-expect";
import { NavigationIcon } from "@suki-web/suki-ui";
import { cleanup, RenderResult } from "@testing-library/react";

import { PageHeader, PageHeaderProps } from ".";
import { componentSetupWithStore } from "../../../hocs";
import { shouldMatchSnapshot } from "../../../helpers/snapshot.spec";

const props: PageHeaderProps = {
  title: "Page title",
  renderActions: () => {
    return (
      <NavigationIcon
        icon="close"
        onClick={() => {
          console.log("");
        }}
        size="xs"
        testId="header-close-button"
      />
    );
  }
};

const setup = componentSetupWithStore((props) => <PageHeader {...props} />);

describe("Page Header component", () => {
  let component: RenderResult;

  afterEach(() => {
    cleanup();
  });

  it("should render successfully", () => {
    component = setup(props);
    const { baseElement } = component;
    expect(baseElement).toBeTruthy();
  });

  it("should have back button", () => {
    component = setup(props);
    const { baseElement } = component;
    expect(baseElement.getElementsByTagName("svg")[0]).toBeTruthy();
  });

  it("should have no add button", () => {
    component = setup({ title: "Page title", hasBackButton: false });
    const { baseElement } = component;
    expect(baseElement.querySelector("button")).toBeFalsy();
  });

  it("should have add button", () => {
    component = setup({ ...props, hasBackButton: false, hasAddButton: true });
    const { baseElement } = component;
    expect(baseElement.querySelector("button")).toBeTruthy();
  });

  it("should have add button with title 'Add button'", () => {
    const newProps = {
      ...props,
      hasAddButton: true,
      renderActions: () => <p>Add button</p>
    };
    component = setup(newProps);
    const { baseElement } = component;
    expect(baseElement.getElementsByTagName("p")[0].innerHTML).toEqual("Add button");
  });

  it("should match snapshot", () => {
    shouldMatchSnapshot(PageHeader, props);
  });
});
