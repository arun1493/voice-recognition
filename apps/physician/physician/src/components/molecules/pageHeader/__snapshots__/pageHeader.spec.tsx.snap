// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Header component should match snapshot 1`] = `
<header
  className="pageHeaderstyled__StyledHeader-sc-1c290ge-0 bbpvdc"
>
  <div
    className="navigationIconstyled__StyledNavigationIcon-lwvkbw-0 imuqWK"
    data-cy="header-close-button"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <button
      className="iconstyled__StyledButton-ifk1iv-0 kgKJBV"
      onClick={[Function]}
    >
      <svg
        height={24}
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.70857 6.29289C7.31805 5.90237 6.68488 5.90237 6.29436 6.29289C5.90383 6.68342 5.90383 7.31658 6.29436 7.70711L10.5865 11.9993L6.2944 16.2914C5.90388 16.6819 5.90388 17.3151 6.2944 17.7056C6.68493 18.0961 7.31809 18.0961 7.70862 17.7056L12.0007 13.4135L16.2928 17.7056C16.6834 18.0961 17.3165 18.0961 17.7071 17.7056C18.0976 17.3151 18.0976 16.6819 17.7071 16.2914L13.4149 11.9993L17.7071 7.70711C18.0976 7.31658 18.0976 6.68342 17.7071 6.29289C17.3166 5.90237 16.6834 5.90237 16.2929 6.29289L12.0007 10.5851L7.70857 6.29289Z"
          fill="#999999"
          xmlns="http://www.w3.org/2000/svg"
        />
      </svg>
    </button>
  </div>
  <p
    className="textstyled__StyledText-sc-1vedkxi-0 ELlxq"
    color="inherit"
    cursor="default"
    onClick={[Function]}
    size="lg"
  >
    Page title
  </p>
</header>
`;
